qiskit/transpiler/oracle_compiler/utils.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Internal utils for Oracle Compiler"""
+
+from qiskit import QuantumCircuit
+from qiskit.circuit.library.standard_gates import ZGate, TGate, SGate, TdgGate, SdgGate, U1Gate, \
+    XGate, HGate, U3Gate
+from .exceptions import OracleCompilerError
+
+
+def tweedledum2qiskit(tweedledum_circuit, name=None, qregs=None):
+    """ Converts a Tweedledum circuit into a Qiskit circuit. A Tweedledum circuit is a
+    dictionary with the following shape:
+        {
+        "num_qubits": 2,
+        "gates": [{
+            "gate": "X",
+            "qubits": [1],
+            "control_qubits": [0],
+            "control_state": "1"
+        }]
+    Args:
+        tweedledum_circuit (dict): Tweedledum circuit.
```

Comment:
Are users supposed to interact with this function? If yes, it might be helpful to add a reference to Tweedledum's circuit documentation here. (or just tweedledum)

Reply:
It should be mostly internal. Do you thin it should be `_tweedledum2qiskit`?

Reply:
It might be a nice function to have if there would be users already knowing Tweedledum and coming to Qiskit, so I think we could leave it public. 🙂

Reply:
Something like this 4832d41db ?